### PR TITLE
LaTeX command improvements and helpers

### PR DESCRIPTION
**Pull request by nspark**
This PR:
- [X] Adds LaTeX commands to simplify and standardize deprecation styling
- [X] Updates `\openshmem` to take an optional specification version argument
- [X] Adds `\CTYPE` command (similar to `\VAR` and `\FUNC`)
- [X] Adds the `\Cstd` command that takes an optional argument; e.g.,` \Cstd` renders as `\textit{C}`, `\Cstd[11]` renders as `\textit{C11}`
- [X] Similarly, update `\Cpp` and `\Fortran`
- [X] Update all references of `\Clang` or `\Celev` to `\Cstd`
- [x] Update existing `\Fortran{}` and `\Cpp{}` invocations to remove `{}`

### CODE REVIEW

**Comment by nspark** [comment]
Here's an example of the deprecation highlighting (taken from #48):

deprecation-highlight

The deprecation highlighting provides two environments `DeprecateBlock` and `DeprecateInline` so that entire paragraphs/blocks and individual sentences, respectively, can be deprecated consistently.

**Comment by jdinan** [comment]
Looks good to me.  Can I get a +1 from another committee member?

**Comment by nspark** [comment]
@jdinan Scope creep! I just pushed a relatively big change to update the C-language macros.

File: content/backmatter.tex [hunk 1]
```diff
@@ -234,7 +234,7 @@ \section{\ac{MPI} Interoperability}
 \openshmem routines can be used to communicate with processes running from the
 same or different executable files, provided that the communication is limited
 to symmetric data objects.  On these systems, static memory such as a
-\Fortran{} common block or \Clang{} global variable, is symmetric between
+\Fortran{} common block or \Cstd global variable, is symmetric between
```
Comment (jdinan) [inline content/backmatter.tex]: Should the existing ```\Fortran{}``` and ```\Cpp{}``` invocations be updated to remove the ```{}```?

Comment (nspark) [inline content/backmatter.tex]: Yes, I think it would be good if we're consistent across the board. I'll make a note to address that.

Comment (jdinan) [inline content/backmatter.tex]: @nspark Did you have a chance to make these changes?

Comment (nspark) [inline content/backmatter.tex]: @jdinan No, not yet, unfortunately.

**Comment by nspark** [comment]
@jdinan Updated the Fortran macros

**Comment by jdinan** [comment]
@nspark We just merged in a few more "C11" changes for you to fix up.  😅

File: content/execution_model.tex [hunk 2]
```diff
@@ -20,7 +20,7 @@
 to the total number of \ac{PE}s minus 1. \ac{PE} identifiers are used for
 \openshmem calls (e.g. to specify \OPR{put} or \OPR{get} routines on symmetric
 data objects, collective synchronization calls) or to dictate a control flow for
-\ac{PE}s using constructs of \Clang{} or \Fortran. The identifiers are fixed for
+\ac{PE}s using constructs of \Cstd or \Fortran. The identifiers are fixed for
```
Comment (manjugv) [inline content/execution_model.tex]: \ac{PE}s -> \acp{PE}

File: content/shmem_collect.tex [hunk 3]
```diff
@@ -56,7 +56,7 @@
     \VAR{pSync} must be of type integer and size \CONST{SHMEM\_COLLECT\_SYNC\_SIZE}.
     If you are using \Fortran, it must be a default integer value.  Every element of
     this array must be initialized with the value \CONST{SHMEM\_SYNC\_VALUE} in
-    \CorCpp{} or \CONST{SHMEM\_SYNC\_VALUE} in \Fortran{} before any of the \ac{PE}s
+    \CorCpp{} or \CONST{SHMEM\_SYNC\_VALUE} in \Fortran before any of the \ac{PE}s
```
Comment (manjugv) [inline content/shmem_collect.tex]: \ac{PE}s -> \acp{PE}

**Comment by manjugv** [comment]
@nspark Besides minor comments I made, the changes look fine. I'm guessing you want to handle new merges to the master, before we merge this commit.

File: content/shmem_collect.tex (hunk 3 above)
Comment (nspark) [inline content/shmem_collect.tex]: @manjugv There are 114 instances of `\ac{PE}s` in the spec. Should I fix them all?

Comment (nspark) [inline content/shmem_collect.tex]: Done.

**Comment by nspark** [comment]
@jdinan I think that this PR is ready to merge.

File: content/shmem_collect.tex (hunk 3 above)
Comment (manjugv) [inline content/shmem_collect.tex]: I think it would be better.

File: content/backmatter.tex [hunk 4]
```diff
@@ -396,13 +396,14 @@ \subsection{\CorCpp: \_my\_pe, \_num\_pes, shmalloc, shfree, shrealloc, shmemali
...
+The \Fortran routines \FUNC{START\_PES}, \FUNC{MY\_PE}, and \FUNC{NUM\_PES}
 were deprecated in order to minimize the API differences from the deprecation
 of \CorCpp routines \FUNC{start\_pes}, \FUNC{\_my\_pe}, and \FUNC{\_num\_pes}.
 
 \subsection{\Fortran: SHMEM\_PUT}
-The \Fortran{} routine \FUNC{SHMEM\_PUT} is defined only for the \Fortran{}
-\ac{API} and is semantically identical to \Fortran{} routines
+The \Fortran function \FUNC{SHMEM\_PUT} is defined only for the \Fortran
+\ac{API} and is semantically identical to \Fortran functions
+
```
Comment (jdinan) [inline content/backmatter.tex]: Is the extra line intentional?

Comment (jdinan) [inline content/backmatter.tex]: @nspark Could you check on the change noted above?  Apart from that, I'm ready to merge.

Comment (nspark) [inline content/backmatter.tex]: @jdinan Good catch. That was a manual-merge error that I introduced.

**Comment by jdinan** [comment]
Thanks!

**Pull request merged**